src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -127,7 +127,13 @@
  * EventNone and state sKeeperStateDisconnected.
  *
  */
-public class ZooKeeper {
+// We suppress the "try" warning here because the close() method's signature
```

Comment:
Good point on avoiding mixing implementation comments with end user facing documents. Though in this case it is safe to use block comments as @eribeiro pointed out because block comments (enclosed in '/* */') will not be part of java doc (enclosed in '/** */'), also use block comments provide better visual consistency style with reset of multi line comments in src file.

It might also be helpful to add to Java doc on ZooKeeper.close so user aware that since 3.5.3 they could optionally use try-with-resource on ZooKeeper object without explicitly invoking close method.